cli/src/main/java/com/devonfw/tools/ide/commandlet/CommandletManager.java

```diff
@@ -42,16 +42,11 @@ public interface CommandletManager {
 
   /**
    * @param name the {@link Commandlet#getName() name} of the requested {@link Commandlet}.
-   * @return the requested {@link Commandlet}.
-   * @throws IllegalArgumentException if not found.
+   * @return the requested {@link Commandlet} or {@code null} if not found.
    */
   default Commandlet getRequiredCommandlet(String name) {
 
-    Commandlet commandlet = getCommandlet(name);
-    if (commandlet == null) {
-      throw new IllegalArgumentException("The commandlet " + name + " could not be found!");
-    }
-    return commandlet;
+    return getCommandlet(name);
```

Comment:
Please revert including JavaDoc changes. It does not make sense to duplicate `getCommandlet(String)` method.
Too much AI vibe coding?
```suggestion
    Commandlet commandlet = getCommandlet(name);
    if (commandlet == null) {
      throw new IllegalArgumentException("The commandlet " + name + " could not be found!");
    }
    return commandlet;
```